core/src/main/java/io/questdb/cairo/mv/FixedOffsetIntervalIterator.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.mv;
+
+import io.questdb.griffin.engine.groupby.TimestampSampler;
+import io.questdb.std.Numbers;
+import org.jetbrains.annotations.NotNull;
+
+public class FixedOffsetIntervalIterator implements SampleByIntervalIterator {
+    private long maxTimestamp;
+    private long minTimestamp;
+    private TimestampSampler sampler;
+    private int step;
+    private long timestampHi;
+    private long timestampLo;
+
+    @Override
+    public long getMaxTimestamp() {
+        return maxTimestamp;
+    }
+
+    @Override
+    public long getMinTimestamp() {
+        return minTimestamp;
+    }
+
+    @Override
+    public int getStep() {
+        return step;
+    }
+
+    @Override
+    public long getTimestampHi() {
+        return timestampHi;
+    }
+
+    @Override
+    public long getTimestampLo() {
+        return timestampLo;
+    }
+
+    @Override
+    public boolean next() {
+        if (timestampHi != maxTimestamp) {
+            timestampLo = timestampHi;
+            for (int i = 0; i < step; i++) {
+                timestampHi = sampler.nextTimestamp(timestampHi);
```

Comment:
as I understand step can be quite large 10M or so. does this loop mean we want min(maxTimestamp, sampler.nextNTimestamp(timestampHi, step)) ?

Reply:
Resolved in 2daccaa and 0f737bb

Reply:
Resolved in 5839d92, but I think the year sampler may need more tests, like the month sampler already has.